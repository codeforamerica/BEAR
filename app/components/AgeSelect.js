// @flow
import React, { Component } from 'react';

type Props = {
  labelText: string,
  minAge: number,
  maxAge: number,
  onAgeSelect: number => void,
  selectedAge: number
};

export default class AgeSelect extends Component<Props> {
  handleNumberSelect = (e: SyntheticEvent<HTMLSelectElement>) => {
    const { onAgeSelect } = this.props;
    console.log('hello', e.currentTarget.selectedOptions[0].text);
    onAgeSelect(parseInt(e.currentTarget.selectedOptions[0].text, 10));
  };

  ageRange = (subjectAgeThreshold: number, maximumAge: number) => {
    const list = [];
    // eslint-disable-next-line no-plusplus
    for (let i = subjectAgeThreshold; i <= maximumAge; i++) {
      list.push(i);
    }
    return list;
  };

  render() {
    const { labelText, minAge, maxAge, selectedAge } = this.props;
    return (
      <div>
        {/* Below keeps failing linter despite having associated control (and passing screenreader) */}
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        {/* eslint-disable-next-line jsx-a11y/label-has-for */}
        <label htmlFor="age-select" className="sr-only">
          {labelText}
        </label>
        <div className="select">
          <select
            className="select__element"
            onChange={this.handleNumberSelect}
            id="age-select"
            value={selectedAge}
          >
            {this.ageRange(minAge, maxAge).map(age => (
              <option key={age} value={age}>
                {age}
              </option>
            ))}
          </select>
        </div>
      </div>
    );
  }
}
