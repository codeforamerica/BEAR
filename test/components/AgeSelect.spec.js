import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import AgeSelect from '../../app/components/AgeSelect';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup() {
  const onAgeSelectSpy = sandbox.spy();
  const component = shallow(
    <AgeSelect onAgeSelect={onAgeSelectSpy} minAge={1} maxAge={3} />
  );
  return {
    component,
    onAgeSelectSpy
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('AgeSelect component', () => {
  it('selecting age should call onAgeSelect with numeric value', () => {
    const { component, onAgeSelectSpy } = setup();
    const ageSelect = component.find('#age-select');
    const fakeEvent = {
      currentTarget: {
        value: '3',
        selectedOptions: [{ text: '3' }]
      }
    };
    ageSelect.simulate('change', fakeEvent);
    expect(onAgeSelectSpy.called).toBe(true);
    expect(onAgeSelectSpy.args[0][0]).toEqual(3);
  });

  it('should populate dropdown with numbers between max and min', () => {
    const { component } = setup();
    const ageSelect = component.find('#age-select');
    const ageOptions = ageSelect.find('option').map(opt => {
      return opt.props().value;
    });
    expect(ageOptions).toEqual([1, 2, 3]);
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
