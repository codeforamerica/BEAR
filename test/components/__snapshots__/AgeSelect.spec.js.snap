// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeSelect component should match exact snapshot 1`] = `
<div>
  <label
    className="sr-only"
    htmlFor="age-select"
  />
  <div
    className="select"
  >
    <select
      className="select__element"
      id="age-select"
      onChange={[Function]}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
    </select>
  </div>
</div>
`;
